tasks/Task.cpp

```diff
@@ -42,8 +61,167 @@ void Task::errorHook()
 void Task::stopHook()
 {
     TaskBase::stopHook();
+    ouster::sensor::sensor_config config;
+    config.operating_mode = ouster::sensor::OperatingMode::OPERATING_STANDBY;
+    if (!sensor::set_config(m_sensor_hostname, config, 0)) {
+        LOG_ERROR_S << "Failed to configure Lidar!" << std::endl;
+    }
+    m_handle.reset();
 }
 void Task::cleanupHook()
 {
     TaskBase::cleanupHook();
 }
+
+sensor::sensor_info Task::getMetadata()
+{
+    auto metadata = sensor::get_metadata(*m_handle);
+    return sensor::parse_metadata(metadata);
+}
+
+LidarScan Task::acquireData()
+{
+    size_t width = m_metadata.format.columns_per_frame;
+    size_t heigth = m_metadata.format.pixels_per_column;
```

Comment:
you don't need this variables, you can pass them straight into Lidarscan scan{}

Reply:
i am using this in another point

Reply:
actually you are not using it inside this function...

Reply:
I am using in the loop to fill the depth_map.distances

Reply:
this is in another function: void Task::convertDataAndWriteOutput(LidarScan& scan)
you create them again at lines 148 and 149.